packages/neptune-extension/src/common/hooks/notebook.ts

```diff
@@ -1,37 +1,18 @@
-
 import React from 'react'
-import { useDispatch, useSelector } from 'react-redux'
+import { useDispatch } from 'react-redux'
 
 import { PlatformNotebook } from 'types/platform'
-
-import { setNotebook } from 'common/state/notebook/actions'
-
-import { getNotebook } from 'common/utils/upload'
+import { fetchNotebook } from "common/state/notebook/actions";
 
 export function loadInitialNotebook(platformNotebook: PlatformNotebook) {
 
   const dispatch = useDispatch()
 
   React.useEffect(() => {
-    async function fetchNotebook() {
-      const metadata = platformNotebook.getMetadata();
-
-      if (metadata.notebookId) {
-        // Load notebook from remote.
-        try {
-          const notebook = await getNotebook(metadata.notebookId)
-          dispatch(setNotebook(notebook))
-        } catch(e) {
-          // 404 or 403
-          dispatch(setNotebook(undefined))
-        }
-      }
-      else {
-        dispatch(setNotebook(undefined))
-      }
+    const metadata = platformNotebook.getMetadata();
+    if (metadata.notebookId) {
+      dispatch(fetchNotebook(metadata.notebookId))
```

Comment:
In this case we cannot use the same error handling as in upload modal because errors are swallowed here and we notify that notebook loading has finished.